cirq-google/cirq_google/calibration/workflow_test.py

```diff
@@ -1342,6 +1380,82 @@ def test_zmake_zeta_chi_gamma_compensation_for_moments_invalid_argument_fails()
         )
 
 
+def test_make_zeta_chi_gamma_compensation_for_moments_imperfect_gates():
+    params_cz_ab = cirq_google.PhasedFSimCharacterization(
+        zeta=0.02, chi=0.05, gamma=0.04, theta=0.0, phi=np.pi
+    )
+    params_cz_cd = cirq_google.PhasedFSimCharacterization(
+        zeta=0.03, chi=0.08, gamma=0.03, theta=0.0, phi=np.pi
+    )
+    params_syc_ab = cirq_google.PhasedFSimCharacterization(
+        zeta=0.01, chi=0.09, gamma=0.02, theta=np.pi / 2, phi=np.pi / 6
+    )
+    params_sqrt_iswap_ac = cirq_google.PhasedFSimCharacterization(
+        zeta=0.05, chi=0.06, gamma=0.07, theta=np.pi / 4, phi=0.0
+    )
+    params_sqrt_iswap_bd = cirq_google.PhasedFSimCharacterization(
+        zeta=0.01, chi=0.02, gamma=0.03, theta=np.pi / 4, phi=0.0
+    )
+
+    a, b, c, d = cirq.LineQubit.range(4)
+    engine_simulator = cirq_google.PhasedFSimEngineSimulator.create_from_dictionary(
+        parameters={
+            (a, b): {
+                cirq.FSimGate(theta=0, phi=np.pi): params_cz_ab,
```

Comment:
Could you use cirq.CZ gate here and below?

Reply:
I can't, because cirq.CZ != FSimGate(theta=0, phi=np.pi), but gate translator converts all gates to FSimGate. So  all gates in this map must be FSimGate, otherwise there will be no match.

Reply:
Yes, you're right. If this code will be used heavily in the future we might think about normalizing all gates to a common form inside the "create_from_dictionary" by using the converter itself. (Let's postpone it for now.)